### PR TITLE
feature state and upgrade/downgrade testing

### PR DESCRIPTION
**Pull request by pohly**
Declaring the entire project as "alpha" is not correct anymore,
several features are considered stable and ready for
production. However, some may still be experimental. We need to
document this in more detail and ensure that we have corresponding tests.

Fixes: https://github.com/intel/pmem-csi/issues/631

### CODE REVIEW

**Comment by pohly** [comment]
TODO:
- [ ] get rid of the forked Kubernetes (issue #722)

**Comment by pohly** [comment]
> should the test really be called "version skew"?

I've added actual skew testing, so the overall name of the new testsuite seems suitable. I kept it.

**Comment by pohly** [comment]
CI testing ran into an "interesting" error:
> 01:13:50.064     pmem-driver: WAITING: ImagePullBackOff - Back-off pulling image "intel/pmem-csi-driver:v0.7.19"

I suspect that was because the release had been tagged, but the image wasn't pushed yet by our CI. Determining which release to test against must be made more intelligent to account for that...

**Comment by pohly** [comment]
The problem I asked about in https://github.com/intel/pmem-csi/pull/637/commits/acfeaf686dad8226bf77b30ff8121139641c0645#r436079052 was causing timeouts in the CI testing. I've reverted that commit.

File: test/e2e/e2e.go [hunk 1]
```diff
@@ -50,9 +52,16 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
...
+	// Therefore we set a fake rate limiter which never throttles.
+	config, err := framework.LoadConfig()
 	if err != nil {
-		framework.Failf("Error loading client: %v", err)
+		framework.Failf("error creating client config: %v", err)
+	}
+	config.RateLimiter = flowcontrol.NewFakeAlwaysRateLimiter()
+	c, err := clientset.NewForConfig(config)
+	if err != nil {
+		framework.Failf("error creating client: %v", err)
```
Comment (pohly) [inline test/e2e/e2e.go:64]: Test run *without* this commit:
https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/version-skew-2/22/pipeline/

Test run *with* it:
https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/version-skew-2/23/pipeline

**Comment by avalluri** [comment]
@pohly We could revert 1a433b7437bf14464d68014aaf032e43243d2e21. I see that is causing the operator to detect default values as changes.

**Comment by pohly** [comment]
> @pohly We could revert 1a433b7. I see that is causing the operator to detect default values as changes.

Let's discuss in https://github.com/intel/pmem-csi/issues/742 - this is not related to this PR.

**Comment by pohly** [comment]
@avalluri : both PR and branch testing have passed. Please review.

The PR contains more than just version skew testing because other things came up while working on this. I can pull out some changes into separate PRs if you want, but others then will cause merge conflicts.

**Review by avalluri — APPROVED** [review]
Looks good to me.

**Pull request merged**